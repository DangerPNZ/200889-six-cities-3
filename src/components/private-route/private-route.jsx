import React from 'react';
import {Route, Redirect} from 'react-router-dom';
import PropTypes from 'prop-types';

const PrivateRoute = ({component, path, redirectPath, isPass}) => (
  <Route exact path={path}>
    {isPass ? component : <Redirect to={redirectPath}/>}
  </Route>
);

PrivateRoute.propTypes = {
  component: PropTypes.element.isRequired,
  path: PropTypes.string.isRequired,
  redirectPath: PropTypes.string.isRequired,
  isPass: PropTypes.bool.isRequired
};

export {PrivateRoute};
