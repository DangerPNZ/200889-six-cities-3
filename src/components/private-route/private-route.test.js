import React from 'react';
import renderer from 'react-test-renderer';
import {PrivateRoute} from './private-route.jsx';
import {BrowserRouter} from 'react-router-dom';

const TestDataValue = {
  ELEMENT: <div></div>,
  PATH: `/component`,
  REDIRECT_PATH: `/redirect`
};

it(`PrivateRoute component structure test with pass access`, () => {
  const tree = renderer
  .create(
      <BrowserRouter>
        <PrivateRoute
          component = {TestDataValue.ELEMENT}
          path = {TestDataValue.PATH}
          redirectPath = {TestDataValue.REDIRECT_PATH}
          isPass = {true}
        />
      </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`PrivateRoute component structure test with pass denied`, () => {
  const tree = renderer
  .create(
      <BrowserRouter>
        <PrivateRoute
          component = {TestDataValue.ELEMENT}
          path = {TestDataValue.PATH}
          redirectPath = {TestDataValue.REDIRECT_PATH}
          isPass = {false}
        />
      </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

