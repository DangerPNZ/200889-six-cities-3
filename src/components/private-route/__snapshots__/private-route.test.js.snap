// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute component structure test with pass access 1`] = `null`;

exports[`PrivateRoute component structure test with pass denied 1`] = `null`;
